src/lavinmqperf/mqtt/throughput.cr

```diff
@@ -0,0 +1,345 @@
+        end
+        @parser.on("-R consumer-rate", "--consumer-rate=number", "Max consume rate (default 0)") do |v|
+          @consume_rate = v.to_i
+        end
+        @parser.on("-j", "--json", "Output result as JSON") do
+          @json_output = true
+        end
+        @parser.on("-z seconds", "--time=seconds", "Only run for X seconds") do |v|
+          @timeout = Time::Span.new(seconds: v.to_i)
+        end
+        @parser.on("-q", "--quiet", "Quiet, only print the summary") do
+          @quiet = true
+        end
+        @parser.on("-C messages", "--pmessages=messages", "Publish max X number of messages") do |v|
+          @pmessages = v.to_i
+        end
+        @parser.on("-D messages", "--cmessages=messages", "Consume max X number of messages") do |v|
+          @cmessages = v.to_i
+        end
+        @parser.on("--random-bodies", "Each message body is random") do
+          @random_bodies = true
+        end
+        @parser.on("--retain", "Set retain flag on published messages") do
+          @retain = true
+        end
+        @parser.on("--clean-session", "Use clean session") do
+          @clean_session = true
+        end
+        @parser.on("-u uri", "--uri=uri", "MQTT broker URI (default mqtt://localhost:1883)") do |v|
+          @uri = URI.parse(v)
+        end
+      end
+
+      @pubs = 0_u64
+      @consumes = 0_u64
+      @stopped = false
+      @counter_lock = Mutex.new
+
+      private def create_client(id : Int32, role : String) : {TCPSocket, LavinMQ::MQTT::IO}
+        host = @uri.host || "localhost"
+        port = @uri.port || 1883
+        user = @uri.user || "guest"
+        password = @uri.password || "guest"
+
+        socket = TCPSocket.new(host, port)
+        socket.keepalive = true
+        socket.tcp_nodelay = false
+        socket.sync = true
+        io = LavinMQ::MQTT::IO.new(socket)
+
+        client_id = "#{role}-#{id}"
+        connect_packet = LavinMQ::MQTT::Connect.new(
+          client_id: client_id,
+          clean_session: @clean_session,
+          keepalive: 30_u16,
+          username: user,
+          password: password.to_slice,
+          will: nil
+        )
+
+        connect_packet.to_io(io)
+
+        response = LavinMQ::MQTT::Packet.from_io(io)
+        unless response.is_a?(LavinMQ::MQTT::Connack) &&
+               response.as(LavinMQ::MQTT::Connack).return_code == LavinMQ::MQTT::Connack::ReturnCode::Accepted
+          socket.close rescue nil
+          raise "Failed to connect: #{response.inspect}"
+        end
+
+        {socket, io}
+      rescue ex : Exception
+        socket.try &.close rescue nil
+        puts "Error creating connection: #{ex.message}"
+        raise ex
+      end
+
+      def run
+        super
+
+        done = WaitGroup.new(@consumers + @publishers)
+        @consumers.times do |i|
+          spawn { reconnect_on_disconnect(done) { consume(i) } }
+        end
+
+        sleep 1.seconds # Give consumers time to connect
```

Comment:
```suggestion
        sleep 1.second # Give consumers time to connect
```

Reply:
Huh, TIL! didn't know there was a `.second`, wonder what the difference from `.seconds` is 😮